### PR TITLE
Make `TrackingBatch` `service_id` an `Option`

### PR DESCRIPTION
**Pull request by dnewh**
This makes the `service_id` field on the `TrackingBatch` struct an
    `Option`. This is in support of work on the batch queuer.

### CODE REVIEW

File: sdk/src/batch_tracking/store/diesel/models.rs [hunk 1]
```diff
@@ -509,8 +514,14 @@ impl
 pub fn make_new_batch_models(batches: &[TrackingBatch]) -> Vec<NewBatchModel> {
     let mut models = Vec::new();
     for batch in batches {
+        let serv_id = if let Some(b) = batch.service_id() {
```
Comment (leebradley) [inline sdk/src/batch_tracking/store/diesel/models.rs:517]: The `b` isn't a great name, but you could avoid it entirely by using
```rust
let serv_id = batch.service_id().unwrap_or(NON_SPLINTER_SERVICE_ID_DEFAULT);
```

File: sdk/src/batch_tracking/store/diesel/models.rs [hunk 2]
```diff
@@ -142,8 +143,12 @@ impl
             Option<SubmissionError>,
         ),
     ) -> Self {
+        let mut serv_id = Some(batch.service_id.to_string());
```
Comment (leebradley) [inline sdk/src/batch_tracking/store/diesel/models.rs:146]: This assignment could all be a single statement without mut

**Review by leebradley — CHANGES_REQUESTED** [review]

File: sdk/src/batch_tracking/store/diesel/models.rs (hunk 1 above)
Comment (dnewh) [reply sdk/src/batch_tracking/store/diesel/models.rs]: updated

File: sdk/src/batch_tracking/store/diesel/models.rs (hunk 2 above)
Comment (dnewh) [reply sdk/src/batch_tracking/store/diesel/models.rs]: updated

**Review by leebradley — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

**Pull request merged**